### PR TITLE
Add unit tests for predictions result transformers

### PR DESCRIPTION
**Pull request by raphkim**
*Issue #, if available:*

*Description of changes:*
* Unit test for AWS Textract result -> Amplify model transformers
* Add more unit tests for AWS Rekognition result -> Amplify model transformers
* Add entities detection configuration unit tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-predictions/src/main/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformers.java [hunk 1]
```diff
@@ -209,7 +210,13 @@ public static BoundedKeyValue fetchKeyValue(@Nullable Block block, @NonNull Map<
                 .build();
     }
 
-    private static Cell fetchTableCell(Block block, Map<String, Block> blockMap) {
+    @VisibleForTesting
```
Comment (jamesonwilliams) [inline aws-predictions/src/main/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformers.java:213]: The `@VisibleForTesting` thing isn't ideal, and is usually used for constructor overloads.

Why not just test `fetchTable`, and keep this is a `private` implementation detail of it? If `fetchTable` passes your test cases, you don't really care about what happens inside this cell-processing routine.

Comment (raphkim) [reply aws-predictions/src/main/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformers.java:213]: I can do that! There was no real reason for me to do it besides breaking up test suites into smaller components

File: aws-predictions/src/test/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformersTest.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                .withGeometry(randomGeometry())
+                .withRelationships(new Relationship()
+                        .withIds(keyTextBlock.getId(), valueBlock.getId()));
+
+        // Construct a map to act as a graph
+        Map<String, Block> blockMap = new HashMap<>();
+        blockMap.put(valueTextBlock.getId(), valueTextBlock);
+        blockMap.put(keyTextBlock.getId(), keyTextBlock);
+        blockMap.put(valueBlock.getId(), valueBlock);
+        blockMap.put(keyBlock.getId(), keyBlock);
```
Comment (jamesonwilliams) [inline aws-predictions/src/test/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformersTest.java:225]: Whenever you duplicate an operation 2-3 times, you should move to a control structure.
```
List<Block> blocks = Arrays.asList(valueTextBlock, keyTextBlock, valueBlock, keyBlock);
Map<String, Block> blocksById = new HashMap<>();
for (Block block : blocks) {
    blocksById.put(block.getId(), block);
}
```
Or:
```
Map<String, Block> blocksById = new HashMap<>();
Observable.fromArray(valueTextBlock, keyTextBlock, valueBlock, keyBlock)
    .blockingForEach(block -> blocksById.put(block.getId(), block));
```

Comment (raphkim) [reply aws-predictions/src/test/java/com/amplifyframework/predictions/aws/adapter/TextractResultTransformersTest.java]: the second solution seems nice! I'll do that

**Pull request merged**